cvmfs/network/download.cc

```diff
@@ -1048,12 +982,21 @@ void DownloadManager::SetUrlOptions(JobInfo *info) {
     url = ReplaceAll(url, "@proxy@", replacement);
   }
 
-  if ((info->destination == kDestinationMem) &&
-      (info->destination_mem.size == 0) &&
-      HasPrefix(url, "file://", false))
-  {
-    info->destination_mem.size = 64*1024;
-    info->destination_mem.data = static_cast<char *>(smalloc(64*1024));
+  // TODO(heretherebedragons) how do we make size_ == 0 generic?
+  // MAYBE have an empty non-functional sink, so that you do not have to check
+  // for kDestinationNone???
```

Comment:
Let's simply use `destination_sink == NULL` as the marker for "HEAD request / no sink"

Reply:
@jblomer having a second look at it
i think we could do here some error handling? != 0 should only be returned when stat64 returns an error -- so i guess this is only the case when the file does not exist or out of memory https://linux.die.net/man/2/stat64

should we do something there or just wait for the error to be caught at a different point?

```c++
int retval = platform_stat(url.c_str(), &stat_buf); 
   if (retval != 0)  { // this should mean here is an error that the file(url) does not exist }
   ```